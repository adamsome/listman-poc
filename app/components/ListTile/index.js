import React from 'react';

import './ListTile.scss'

const ListTile = (props) => {
  return (
    <a className="list-tile box">
      <span className="list-tile__title-wrapper">
        <span className="list-tile__title title is-4">
          {props.title}
        </span>
      </span>
    </a>
  )
}

export default ListTile
