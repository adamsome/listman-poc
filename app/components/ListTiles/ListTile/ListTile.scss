.list-tile {
  padding-bottom: 0.5rem;
}
.list-tile__title-wrapper {
  display: block;
  min-height: 60px;
}

.list-tile__title {
  display: inline-block;
  overflow: hidden;
  max-height: 48px;
}

