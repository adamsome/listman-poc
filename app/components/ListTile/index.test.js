import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ListTile from './index'

let wrapper
beforeEach(() => {
  wrapper = shallow(<ListTile title='ListTile Title' />)
})

it('shows title', () => {
  expect(wrapper.type()).toEqual('a')
  expect(wrapper.contains('ListTile Title')).toBe(true)
  expect(toJson(wrapper)).toMatchSnapshot()
})

