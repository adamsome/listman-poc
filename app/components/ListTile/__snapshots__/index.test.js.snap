exports[`test shows title 1`] = `
<a
  className="list-tile box">
  <span
    className="list-tile__title-wrapper">
    <span
      className="list-tile__title title is-4">
      ListTile Title
    </span>
  </span>
</a>
`;
